### PR TITLE
Add unit-test layer to relay-server

### PR DESCRIPTION
**Pull request by robotoer**
## Summary

The `relay-server/` subproject previously had no unit tests — only the e2e suite in `e2e/integration/` exercised it, and exclusively via a live libp2p stack. This PR extracts the pure config/policy logic out of `relay-server/src/index.ts` so it can be unit-tested in isolation, then adds jest + ts-jest and 47 tests covering it.

### Extracted helpers

- **`relay-server/src/config.ts`** — `loadConfig(env)` translates `WS_PORT`, `TCP_PORT`, `WS_LISTEN`, `TCP_LISTEN`, `ENABLE_IPV6`, `WS_LISTEN_V6`, `TCP_LISTEN_V6`, `DOCUMENT_PUBLISH_PATH`, `TOPIC_ALLOWLIST`, `MAX_AUTO_TOPICS`, and `EXTRA_TOPICS` into a typed `RelayConfig`. `listenAddresses(config)` computes the bound multiaddr list. Defaults match the prior inline behaviour exactly — this is a pure re-shuffle.
- **`relay-server/src/topic-policy.ts`** — `shouldAutoSubscribe(topic, policy)` is a pure function returning `{action:'subscribe'}` or `{action:'skip', reason}` where `reason` is one of `AlreadyTracked`, `SystemTopic`, `NotInAllowlist`, or `CapReached`. `index.ts` keeps ownership of the live `trackedTopics`/`autoTopics` `Set`s and passes them in via `isTracked` / `autoTopicCount`. The libp2p event-listener now just calls the helper and logs.

`index.ts` is now thin: load config, wire libp2p, hand subscription decisions to the helper. No runtime behaviour change.

### Tooling

- Adds `jest`, `ts-jest`, `@types/jest` to `relay-server/devDependencies` (pinned to `^29.2.5`, matching the rest of the repo).
- Adds a `test` script (`"jest"`) and a `jest.config.cjs`.
- The package is ESM (`"type":"module"`) but jest+ts-jest is simpler in CommonJS, so a `tsconfig.test.json` overrides `module` -> `CommonJS` for the test transform only. The production build still emits ESM. A `moduleNameMapper` rewrites the `.js` relative-import suffixes used by the runtime sources so they resolve under jest.
- Excludes `*.test.ts` from the production build via `tsconfig.exclude` so `dist/` stays clean.
- `relay-server/` is **not** part of the yarn workspace root, so the lockfile churn is isolated.

### Test coverage (47 tests)

- **config.test.ts** — defaults, port/listen overrides, `ENABLE_IPV6` semantics (only `"1"` enables, everything else disables), allowlist CSV parsing (single, multi, whitespace, empty, whitespace-only), `MAX_AUTO_TOPICS` fallback for non-numeric/empty/zero/negative, `parseInt` leading-digit semantics, `EXTRA_TOPICS` parsing, `listenAddresses` with and without IPv6.
- **topic-policy.test.ts** — open-mode subscribes; allowlist hit/miss/prefix-vs-equality; cap edge cases (below, at, above); tracked-topic short-circuit; system-topic rejection (`_`, `floodsub:`, caller-overridden prefixes); rejection-precedence pinning so a future refactor can't silently shuffle the gates.

## Test plan

- [x] `yarn install` in `relay-server/` (touches only `relay-server/yarn.lock`)
- [x] `yarn test` in `relay-server/` — 47 tests pass
- [x] `yarn build` in `relay-server/` — `tsc` succeeds, `dist/` contains only `config.{js,d.ts}`, `index.{js,d.ts}`, `topic-policy.{js,d.ts}` (no test artefacts)
- [ ] e2e integration suite still passes (unchanged runtime behaviour, but worth a smoke test)

## Summary by CodeRabbit

* **New Features**
  * Configurable relay server driven by environment variables: ports, listen addresses, document paths, topic allowlist, caps, and extra topics.
  * New auto-subscribe policy with clear rules for system topics, tracked topics, allowlist, and cap enforcement; IPv6 gating supported.

* **Tests**
  * Added comprehensive tests covering configuration loading, listen-address behavior, and auto-subscribe decision logic.

* **Chores**
  * Test tooling and configs updated to run TypeScript tests and a project-level test script added.

[Review Change Stack](https://app.coderabbit.ai/change-stack/swarmbase/swarmbase/pull/283)

### CODE REVIEW

File: relay-server/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * string — i.e. the operator hasn't expressed an opinion, so callers should
+ * treat this as "open mode".
+ *
+ * Returns `[]` when the env var is set to a non-empty value that
+ * nonetheless parses to zero usable entries (e.g. "," or "   "). This
+ * matches the historical inline behaviour where any non-empty string
+ * produced an array (possibly empty), and crucially keeps a misconfigured
+ * allowlist in "closed mode" rather than silently flipping it open.
+ */
+function parseCsv(value: string | undefined): string[] | null {
```
[the remark this answers is not in the document]
Comment (robotoer) [reply relay-server/src/config.ts:78]: The r2 fix already addresses this: `parseCsv()` now returns `null` only when the env var is `undefined` or `''` (truly unset/empty). For non-empty values that parse to zero entries (`','`, `'   '`), it returns `[]` — the legacy closed-mode semantics. The runtime behaviour matches pre-PR behaviour for those edge cases. See lines 78-86 of the current config.ts.

File: relay-server/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+/** Default plain-TCP port. */
+export const DEFAULT_TCP_PORT = '9002'
+
+/** Default document publish path (matches collabswarm-config.ts default). */
+export const DEFAULT_DOCUMENT_PUBLISH_PATH = '/documents'
+
+/**
+ * Topic prefixes that are treated as system/internal and should never be
+ * auto-subscribed. This module is the canonical definition; `topic-policy.ts`
+ * imports this constant rather than redeclaring it, so there's no duplication.
```
[the remark this answers is not in the document]
Comment (robotoer) [reply relay-server/src/config.ts:26]: This was already addressed in r2. The current comment (lines 23-27) reads: "This module is the canonical definition; `topic-policy.ts` imports this constant rather than redeclaring it, so there's no duplication." That matches your requested wording — this r3 comment looks like a stale repeat from r2.
